### PR TITLE
Add support for uint32 bind var + test

### PR DESCRIPTION
**Pull request by akenneth**
Signed-off-by: Alon Kenneth <11458012+akenneth@users.noreply.github.com>

## Description

Add missing support for `uint32` bindvars mapping.

## Related Issue(s)

* Fixes https://github.com/vitessio/vitess/issues/9756

## Checklist
- [ ] Should this PR be backported? **- akenneth: I don't know!**
- [x] Tests were added or are not required
- [x] Documentation was added or is not required - not required i think?

### CODE REVIEW

File: go/vt/vtgate/evalengine/eval_result.go [hunk 1]
```diff
@@ -730,6 +730,12 @@ func (er *EvalResult) setBindVar1(typ sqltypes.Type, value []byte, collation col
 		}
 		// TODO: type32
 		er.setInt64(ival)
+	case sqltypes.Uint32:
+		ival, err := strconv.ParseUint(string(value), 10, 32)
+		if err != nil {
+			ival = 0
```
Comment (vmg) [inline go/vt/vtgate/evalengine/eval_result.go:736]: You must return an error here just like the other cases.

**Review by vmg — CHANGES_REQUESTED** [review]
Thanks for the contribution! Can you fix then error return?

File: go/vt/vtgate/evalengine/eval_result.go (hunk 1 above)
Comment (akenneth) [reply go/vt/vtgate/evalengine/eval_result.go]: Thanks @vmg! I updated to `throwEvalError(err)` 

Please note that the other int cases are not throwing an error in the current state. For example:
https://github.com/vitessio/vitess/pull/9757/files#diff-8bd3f97165c3b882919fff81a641216b78e2c23ba08ca3d0e2d8dab3574bc72aL735-L737

Maybe it makes sense to change all of them to throw?

Comment (vmg) [reply go/vt/vtgate/evalengine/eval_result.go]: Huh, I see now that we have inconsistent behavior there. I'll fix it in a separate PR. Thanks for the heads up!

**Review by vmg — APPROVED** [review]

**Pull request merged**